cmd/util/app.go

```diff
@@ -274,19 +274,19 @@ func SetAppSpecOptions(flags *pflag.FlagSet, spec *argoappv1.ApplicationSpec, ap
 		}
 	})
 	if flags.Changed("auto-prune") {
-		if spec.SyncPolicy == nil || !spec.SyncPolicy.IsAutomatedSyncEnabled() {
+		if spec.SyncPolicy == nil || spec.SyncPolicy.Automated == nil {
```

Comment:
Why does any of these `IsAutomatedSyncEnabled()` have to be changed? The definition looks OK IMO

```
// IsAutomatedSyncEnabled checks if the automated sync is enabled or disabled
func (p *SyncPolicy) IsAutomatedSyncEnabled() bool {
	if p.Automated != nil && (p.Automated.Enabled == nil || *p.Automated.Enabled) {
		return true
	}
	return false
}
```

Reply:
Auto sync is disabled if any of the two conditions are true:

* automated is `nil`
* **OR** if automated is not `nil` **AND** `enabled` has been set to `false`